File: tutorial/code/python/switch.py
import zio
import time

agent = zio.Agent("/dev/ttyUSB0")
gpio = zio.GPIO(agent)
switch_pin = 0 # <1>

while True:
    state = gpio.read_input_pin(switch_pin) # <2>
    if state:
        print "Switch Off"
    else:
	print "Switch On"

    time.sleep(0.5)

File: tutorial/code/java/Switch.java
import com.zilogic.zio.*;

class Switch {
    public static void main(String args[]) 
	throws ProtocolException, InterruptedException {

	Agent agent = new Agent("/dev/ttyUSB0");
	GPIO gpio = new GPIO(agent);
	int switchPin = 0; // <1>
	
	while (true) {
	    int state;

	    state = gpio.readInputPin(switchPin); // <2>
	    if (state == 1) {
		System.out.println("Switch Off");
	    } else {
		System.out.println("Switch On");
	    }

	    Thread.sleep(500);
	}
    }
}
